api/envoy/config/wasm/v2/wasm.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package envoy.config.wasm.v2;
+
+option java_outer_classname = "WasmProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.config.wasm.v2";
+
+import "envoy/api/v2/core/base.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Wasm service]
+
+// Configuration for a Wasm VM.
+// [#next-free-field: 6]
+message VmConfig {
+  // An ID which will be used along with a hash of the wasm code (or null_vm_id) to determine which
+  // VM will be used for the plugin. All plugins which use the same vm_id and code will use the same
+  // VM. May be left blank.
+  string vm_id = 1;
+
+  // The Wasm runtime type (see source/extensions/common/wasm/well_known_names.h).
+  string runtime = 2;
+
+  // The Wasm code that Envoy will execute.
+  api.v2.core.AsyncDataSource code = 3;
+
+  // The Wasm configuration string used on initialization of a new VM (proxy_onStart).
+  string configuration = 4;
+
+  // Allow the wasm file to include pre-compiled code on VMs which support it.
+  bool allow_precompiled = 5;
+}
+
+// Base Configuration for Wasm Plugins e.g. filters and services.
+message PluginConfig {
+  // A unique name for a filters/services in a VM for use in identifying the filter/service if
+  // multiple filters/services are handled by the same vm_id and root_id and for logging/debugging.
+  string name = 1;
+
+  // A unique ID for a set of filters/services in a VM which will share a RootContext and Contexts
```

Comment:
All docs in the API either should be consumable standalone or should have `:ref:` to something that explains concepts. At this point, the reader has no idea about root context or event contexts.

Reply:
@PiotrSikora will be adding the ABI doc and we will reference it here.

Reply:
I would prefer that the ABI doc lands first or we keep track with some annotations of every place we're missing these links.

Reply:
@PiotrSikora Please attach ABI doc here.